File: recipe-main-service/src/main/java/com/abc/recipemainservice/domain/Ingredient.java
package com.abc.recipemainservice.domain;

import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;

import javax.persistence.*;
import java.math.BigDecimal;

@Data
@Entity
@ApiModel(description = "Ingredient")
public class Ingredient {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @ApiModelProperty(notes = "unique identifier")
    private Long id;

    @ApiModelProperty(notes = "description of ingredient")
    private String description;

    @ApiModelProperty(notes = "amount of amount")
    private BigDecimal amount;

    @ManyToOne
    private Recipe recipe;

    @OneToOne(fetch = FetchType.EAGER)
    @ApiModelProperty(notes = "unit of measure")
    private UnitOfMeasure unitOfMeasure;
}

File: recipe-react/src/components/Ingredients/Ingredient/Ingredient.js
import React from 'react';

const ingredient = (props) => {
    return (
        <tr>
            <td>{props.name}</td>
            <td>{props.description}</td>
            <td>{props.unitOfMeasure.uom}</td>
            <td>{props.amount}</td>
        </tr>
    )
};

export default ingredient;